clang/lib/CIR/CodeGen/CIRGenBuiltin.cpp

```diff
@@ -302,6 +302,678 @@ RValue CIRGenFunction::buildRotate(const CallExpr *E, bool IsRotateRight) {
   return RValue::get(r);
 }
 
+/// Get builtin name for unhandled BuiltinID in the function
+/// `CIRGenFunction::buildBuiltinExpr`. Thoese builtin names are handled
+/// in OG code `CodeGenFunction::EmitBuiltinExpr`.
+/// NOTE: Every time we implement handling of a case in
+/// the function `CIRGenFunction::buildBuiltinExpr`, it should be remove here.
+static std::string getUnhandledBuiltinName(unsigned builtinID) {
+  switch (builtinID) {
+  default:
+    return std::string();
+  case Builtin::BI__builtin_acos:
+    return "Builtin::BI__builtin_acos";
+  case Builtin::BI__builtin_acosf:
```

Comment:
When you did similar for Aarch64 my feedback was to make those inline (without a function that returns the name), and you did implement like that. The idea is to get an accurate account of source line of failure, I'm confused on why you took a different approach here.

Reply:
Example: https://github.com/llvm/clangir/blob/81c043e19297585042b742ba87f924f31164a74e/clang/lib/CIR/CodeGen/CIRGenBuiltinAArch64.cpp#L2410

Reply:
I see, sounds like better approach!
Because if we do have them inline instead of returning the name, not only we can have source line of failures (and name can be known by looking up source code), we would have a skeleton that we can fill when we have implementation ready. The only con with this alternative is every time we change the code, failing line number would change, but I think it might be Ok.

I'll  change this PR to use this approach.

Reply:
exactly, good point on the maintain bits too